test/evergreen/msan_suppression_wrappers.c

```diff
@@ -0,0 +1,58 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <sys/stat.h>
+#include "string.h"
+
+/*
+ * FIXME-WT-13103 If you add this file to the list of compiled sources and build with the
+ * `-Wl,--wrap=XXX` linker flag the compiler will wrap calls to function XXX with the respective
+ * __wrap_XXX function defined below. We need this because MSan reports false positives when setting
+ * memory via *stat() calls, so we've defined __wrap_*stat functions that first zero the memory to
+ * suppress the false positive. Once WiredTiger's minimum supported LLVM version is 14 this fix
+ * (https://github.com/llvm/llvm-project/commit/4e1a6c07052b466a2a1cd0c3ff150e4e89a6d87a) is
+ * available and this file can be deleted.
+ */
+
+extern int __real_stat(const char *path, struct stat *buf);
+extern int __real_fstat(int fd, struct stat *statbuf);
+
+/*
+ * MSan is only supported on Clang and these __wrap_* functions should only be used when WiredTiger
+ * is built with MSan.
+ */
+#ifndef __clang__
+#error "msan_suppression_wrappers.c should only be compiled with Clang"
+#elif !__has_feature(memory_sanitizer)
+#error "msan_suppression_wrappers.c should only be compiled with MSan"
+#endif
+
+int __wrap_stat(const char *path, struct stat *buf);
```

Comment:
I don't understand why you declare these prototypes in this file.

Reply:
This was compiler appeasement on my end as without them we hit `-Wmissing-prototypes` warnings. 
I used to have a separate .h file for the prototypes but folded everything into the .c file to keep things compact.